core/lib/snapshots_applier/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+            l1_batch_number,
+            snapshot.storage_logs_chunks.len()
+        );
+
+        let miniblock_root_hash = main_node_client
+            .miniblock_hash(snapshot.miniblock_number)
+            .await
+            .map_err(|err| Fatal(err.into()))?
+            .ok_or_else(|| Fatal(Error::msg("Miniblock is missing")))?;
+
+        let chunk_ids = snapshot
+            .storage_logs_chunks
+            .iter()
+            .map(|x| x.chunk_id)
+            .collect();
+        Ok(SnapshotRecoveryStatus {
+            l1_batch_number,
+            l1_batch_root_hash: snapshot.last_l1_batch_with_metadata.metadata.root_hash,
+            miniblock_number: snapshot.miniblock_number,
+            miniblock_root_hash,
+            storage_logs_chunks_ids_already_processed: vec![],
+            storage_logs_chunks_ids_to_process: chunk_ids,
+        })
+    }
+
+    async fn recover_factory_deps(
+        &mut self,
+        storage: &mut StorageProcessor<'_>,
+    ) -> anyhow::Result<(), SnapshotApplierError> {
+        let factory_deps: SnapshotFactoryDependencies = self
+            .blob_store
+            .get(self.applied_snapshot_status.l1_batch_number)
+            .await?;
+
+        let all_deps_hashmap: HashMap<H256, Vec<u8>> = factory_deps
+            .factory_deps
+            .into_iter()
+            .map(|dep| (hash_bytecode(&dep.bytecode.0), dep.bytecode.0))
+            .collect();
+        storage
+            .storage_dal()
+            .insert_factory_deps(
+                self.applied_snapshot_status.miniblock_number,
+                &all_deps_hashmap,
+            )
+            .await?;
+        Ok(())
+    }
+    async fn insert_initial_writes_chunk(
+        &mut self,
+        storage_logs: &[SnapshotStorageLog],
+        storage: &mut StorageProcessor<'_>,
+    ) -> Result<(), SnapshotApplierError> {
+        tracing::info!("Loading {} storage logs into postgres", storage_logs.len());
+        storage
+            .storage_logs_dedup_dal()
+            .insert_initial_writes_from_snapshot(&storage_logs)
+            .await?;
+        Ok(())
+    }
+    async fn insert_storage_logs_chunk(
+        &mut self,
+        storage_logs: &[SnapshotStorageLog],
+        storage: &mut StorageProcessor<'_>,
+    ) -> Result<(), SnapshotApplierError> {
+        storage
+            .storage_logs_dal()
+            .insert_storage_logs_from_snapshot(
+                self.applied_snapshot_status.miniblock_number,
+                storage_logs,
+            )
+            .await?;
+        Ok(())
+    }
+
+    async fn recover_storage_logs_single_chunk(
+        &mut self,
+        chunk_id: u64,
+    ) -> Result<(), SnapshotApplierError> {
+        let mut storage = self
+            .connection_pool
+            .access_storage_tagged("snapshots_applier")
+            .await?;
+        let mut storage = storage.start_transaction().await?;
```

Comment:
Dumb question: What's the reason for connecting to the DB and opening a transaction before getting data from the blob store?

Reply:
I've used a convention, where opening transaction is the first thing done in a method without deeper thought, I see that moving it after acesing blob store has some benefits, but I'm not yet convinced it's a meaningful optimization in a non-high-load-server situations or I'm missing something

Reply:
update: I've moved it when adding metrics